test/e2e/redundancy/gateway_failover.go

```diff
@@ -137,118 +136,119 @@ func defaultEndpointType() tcp.EndpointType {
+
+	By("Verify active and passive gateway pods exist on active gateway node")
+	var initialActiveGatewayPod string
+	var initialActiveGatewayNode string
+	var initialPassiveGatewayPod string
+	var initialPassiveGatewayNode string
+
+	initialGwPods := f.AwaitPodsByAppLabel(framework.ClusterIndex(primaryCluster), framework.SubmarinerGateway, framework.TestContext.SubmarinerNamespace, 2)
+	for i := range initialGwPods.Items {
+		if initialGwPods.Items[i].Labels[gatewayStatusLabel] == gatewayStatusActive {
+			initialActiveGatewayPod = initialGwPods.Items[i].Name
+			initialActiveGatewayNode = initialGwPods.Items[i].Labels[gatewayNodeLabel]
+			By(fmt.Sprintf("Found active pod %q on gateway node %q", initialActiveGatewayPod, initialActiveGatewayNode))
+		} else if initialGwPods.Items[i].Labels[gatewayStatusLabel] == gatewayStatusPassive {
+			initialPassiveGatewayPod = initialGwPods.Items[i].Name
+			initialPassiveGatewayNode = initialGwPods.Items[i].Labels[gatewayNodeLabel]
+			By(fmt.Sprintf("Found passive pod %q on gateway node %q", initialPassiveGatewayPod, initialPassiveGatewayNode))
+		}
+	}
+	Expect(initialActiveGatewayPod).ToNot(BeEmpty(), fmt.Sprintf("Expected at least one active gateway pod on %q", clusterAName))
+	Expect(initialActiveGatewayNode).ToNot(BeEmpty(), fmt.Sprintf("Failed to detect an active gateway node on %q", clusterAName))
+	Expect(initialPassiveGatewayPod).ToNot(BeEmpty(), fmt.Sprintf("Expected at least one passive gateway pod on %q", clusterAName))
+	Expect(initialPassiveGatewayNode).ToNot(BeEmpty(), fmt.Sprintf("Failed to detect passive gateway node on %q", clusterAName))
 
-	f.AwaitGatewayFullyConnected(framework.ClusterA, resource.EnsureValidName(initialGatewayNode.Name))
-	gwPassive := f.AwaitGatewayWithStatus(framework.ClusterA, initialNonGatewayNode.Name, subv1.HAStatusPassive)
+	By(fmt.Sprintf("Ensuring that passive gateway node %q has no connections", initialPassiveGatewayNode))
+	gwPassive := f.AwaitGatewayWithStatus(framework.ClusterIndex(primaryCluster), initialPassiveGatewayNode, subv1.HAStatusPassive)
 	Expect(gwPassive.Status.Connections).To(BeEmpty(), "The passive gateway must have no connections")
 
-	By("Checking the existing pods have updated HA state labels as active & passive")
-
-	gwPods := f.AwaitPodsByAppLabel(framework.ClusterA, framework.SubmarinerGateway, framework.TestContext.SubmarinerNamespace, 2)
+	submEndpoint := f.AwaitSubmarinerEndpoint(framework.ClusterIndex(primaryCluster), subFramework.NoopCheckEndpoint)
+	By(fmt.Sprintf("Found submariner endpoint for %q: %#v", clusterAName, submEndpoint))
 
-	for i := range gwPods.Items {
-		if gwPods.Items[i].Labels[gatewayNodeLabel] == initialGatewayNode.Name {
-			Expect(gwPods.Items[i].Labels[gatewayStatusLabel]).To(Equal(gatewayStatusActive))
-		} else {
-			Expect(gwPods.Items[i].Labels[gatewayStatusLabel]).To(Equal(gatewayStatusPassive))
-		}
+	By(fmt.Sprintf("Crash the gateway node - %q", initialActiveGatewayNode))
```

Comment:
We may not actually crash the node 😄 

```suggestion
	By("Performing fail-over to passive gateway")
```

Reply:
ack :)

Reply:
Done